actionpack/test/controller/request_forgery_protection_test.rb

```diff
@@ -304,6 +304,50 @@ def test_should_allow_put_with_token_in_header
     assert_not_blocked { put :index }
   end
 
+  def test_should_allow_post_with_origin_checking_and_correct_origin
+    ActionController::Base.forgery_protection_origin_check = true
+    begin
+      session[:_csrf_token] = @token
+      @controller.stub :form_authenticity_token, @token do
+        assert_not_blocked do
+          @request.set_header 'HTTP_ORIGIN', 'http://test.host'
+          post :index, params: { custom_authenticity_token: @token }
+        end
+      end
+    ensure
+      ActionController::Base.forgery_protection_origin_check = false
+    end
+  end
+
+  def test_should_allow_post_with_origin_checking_and_no_origin
+    ActionController::Base.forgery_protection_origin_check = true
+    begin
+      session[:_csrf_token] = @token
+      @controller.stub :form_authenticity_token, @token do
+        assert_not_blocked do
+          post :index, params: { custom_authenticity_token: @token }
+        end
+      end
+    ensure
+      ActionController::Base.forgery_protection_origin_check = false
+    end
+  end
+
+  def test_should_block_post_with_origin_checking_and_wrong_origin
+    ActionController::Base.forgery_protection_origin_check = true
+    begin
+      session[:_csrf_token] = @token
+      @controller.stub :form_authenticity_token, @token do
+        assert_blocked do
+          @request.set_header 'HTTP_ORIGIN', 'http://bad.host'
+          post :index, params: { custom_authenticity_token: @token }
+        end
+      end
+    ensure
+      ActionController::Base.forgery_protection_origin_check = false
+    end
+  end
+
```

Comment:
Agree on the convenience method, could also improve on that by capturing the original value of `forgery_protection_origin_check` and then setting that back in the ensure. That way we don't rely on the default value.